### DISCUSSION TITLE
默认的设置一定要用代理节点么?

### DISCUSSION THREAD

**Discussion by motiyb** [Openwrt-Passwall/openwrt-passwall2#197] [General]
我的想法是所有的流量都直连，只有特别指定的几类网站走代理 (主要是因为穷，走多了代理流量容易爆....)，但看到的教程都说是 默认走代理，这样是不是就变成没有命中规则的网站都会走代理，比如有些没有被墙的国外网站也会走到代理。

所以我想咨询下各位大佬下面的设置是不是能达到说只有Netflix和proxy走代理，还是说有其他坑我还没注意到的... 请赐教，谢谢！

![image](https://user-images.githubusercontent.com/87536739/209900643-6b3db67e-c699-400e-a314-18d046e728a7.png)

**Comment by wangdaye078** [comment]
这样没问题啊，默认改代理，就是一些小网站，但是不支持直连的，可以直接打开，而默认直连的话，你得手动加到代理列表里去，但是那样应该更安全。没有被墙的国外网站也会走到代理不是大问题，一些没有被标记的国内网站也会走代理才是问题。这种流量用XTLS容易被GFW识别。

**Comment by Travel2Here** [comment]
除了你要走代理的配上节点 其他都选关闭 这样能少些匹配过程

**Reply by etsam** [reply]
请问 要是默认选直连，代理选节点，用 https://browserleaks.com/dns   网站测试会识别到本地IP，并且有CN国旗，请问这算DNS泄露吗

**Reply by Deleted-Account** [reply]
你自己把这网站加入代理试试？

**Reply by etsam** [reply]
那这样会不会成了掩耳盗铃，只是强制把这个网站走代理。
另外  https://ipleak.net   用这个网站测试又没有泄露，不明白原理，一头雾水

**Reply by 2eronia** [reply]
有没有可能， ipleak.net 在你设置的规则中有，browserleaks.com 在规则中没有？

我这边也是类似图片上的设置，ipleak 是有国旗的。
加入 domain:ipleak.net 才没有。

掩耳盗铃？估计不是哦。
DNS leak 在我的理解看来，是域名没命中代理导致的。
比如用IPifNonMatch，某个网址不在订阅的列表中，但是解析出来的IP命中了走代理，那这个网站会让设置的国内国外DNS同时解析，才造成DNS leak。
IPOnDemand 则是所有的网址第一次访问，都通过国内国外DNS解析一遍，并把结果缓存起来以确定将来走代理还是直连。
如果是AsIs/IPifNonMatch这种基于网址分流，且域名命中了规则，那就不存在 DNS leak。

**Reply by etsam** [reply]
感谢详细解析
